include/deal.II/non_matching/mapping_info.h

```diff
@@ -1464,6 +1496,283 @@ namespace NonMatching
 
 
 
+  template <int dim, int spacedim, typename Number>
+  template <typename CellIteratorType>
+  void
+  MappingInfo<dim, spacedim, Number>::reinit_faces(
+    const std::vector<std::pair<CellIteratorType, unsigned int>>
+                                           &face_iterator_range_interior,
+    const std::vector<Quadrature<dim - 1>> &quadrature_vector)
+  {
+    clear();
+
+    do_cell_index_compression = false;
+
+    Assert(additional_data.store_cells == false, ExcNotImplemented());
+
+
+    const unsigned int n_faces = quadrature_vector.size();
+    AssertDimension(n_faces,
+                    std::distance(face_iterator_range_interior.begin(),
+                                  face_iterator_range_interior.end()));
+
+    n_q_points_unvectorized.reserve(n_faces);
+
+    cell_type.reserve(n_faces);
+    face_number.reserve(n_faces);
+
+    // fill unit points index offset vector
+    unit_points_index.reserve(n_faces + 1);
+    unit_points_index.push_back(0);
+    data_index_offsets.reserve(n_faces + 1);
+    data_index_offsets.push_back(0);
+    for (const auto &quadrature : quadrature_vector)
+      {
+        const unsigned int n_points = quadrature.size();
+        n_q_points_unvectorized.push_back(n_points);
+
+        const unsigned int n_q_points =
+          compute_n_q_points<VectorizedArrayType>(n_points);
+        unit_points_index.push_back(unit_points_index.back() + n_q_points);
+
+        const unsigned int n_q_points_data =
+          compute_n_q_points<Number>(n_points);
+        data_index_offsets.push_back(data_index_offsets.back() +
+                                     n_q_points_data);
+      }
+
+    const unsigned int n_unit_points = unit_points_index.back();
+    const unsigned int n_data_points = data_index_offsets.back();
+
+    // resize data vectors
+    resize_unit_points(n_unit_points);
+    resize_unit_points_faces(n_unit_points);
+    resize_data_fields(n_data_points, true);
+
+    std::array<MappingData, 2> mapping_data;
+    std::array<MappingData, 2> mapping_data_previous_cell;
+    std::array<MappingData, 2> mapping_data_first;
+    bool                       first_set            = false;
+    unsigned int               size_compressed_data = 0;
+    unsigned int               face_index           = 0;
+    QProjector<dim>            q_projector;
+    for (const auto &cell_and_f : face_iterator_range_interior)
+      {
+        const auto &quadrature_on_face = quadrature_vector[face_index];
+        const bool  empty              = quadrature_on_face.empty();
+
+        // get interior cell and face number
+        const auto &cell_m = cell_and_f.first;
+        const auto  f_m    = cell_and_f.second;
+
+        // get exterior cell and face number
+        const auto &cell_p =
+          cell_m->at_boundary(f_m) ? cell_m : cell_m->neighbor(f_m);
+        const auto f_p =
+          cell_m->at_boundary(f_m) ? f_m : cell_m->neighbor_of_neighbor(f_m);
```

Comment:
This would be the place to add an assert to notify the user that the function is only working for structured meshes.

Reply:
The assert is added in d002a6010323246962fe6b2ec15c10f790facc80